node/channel.js

```diff
@@ -453,20 +454,37 @@ TChannel.prototype.makeSubChannel = function makeSubChannel(options) {
     return chan;
 };
 
-TChannel.prototype.listen = function listen(port, host, callback) {
+TChannel.prototype.listen = function listen(port, host, opts, callback) {
     // Note:
     // - 0 is a valid port number, indicating that the system must assign an
     //   available ephemeral port
     // - 127.0.0.1 is a valid host, primarily for testing
+    // - opts supports:
+    //   {fd: X} - listen will attach to the existing socket, and host/port
+    //   remain required, but become only descriptive. See also:
+    //   https://nodejs.org/api/net.html#net_server_listen_handle_callback
     var self = this;
+    if (opts && typeof opts === 'function') {
+        callback = opts;
+        opts = {};
+    }
+    opts = opts || {};
     assert(!self.listened, 'TChannel can only listen once');
     assert(typeof host === 'string', 'TChannel requires host argument');
     assert(typeof port === 'number', 'TChannel must listen with numeric port');
+    assert(!('fd' in opts) || typeof opts.fd === 'number', 'TChannel listen opts.fd must be numeric');
```

Comment:
Please say `typeof opts.fd === 'number'`. We avoid `in` since it deoptimizes the surrounding function.